lib/forwarder.js

```diff
@@ -38,9 +69,24 @@ export async function forwardToDaemon(resolver, config) {
     );
   }
 
-  const socket = net.connect(config.port, '127.0.0.1');
+  let socket;
+  try {
+    socket = await createSocket(true);
+  } catch (/** @type {any} */ err) {
+    if (err['code'] === 'ECONNREFUSED') {
+      console.error(`eslint_d: ${err} - removing config`);
+      await removeConfig(resolver);
+    } else {
+      console.error(`eslint_d: ${err}`);
+    }
+    // eslint-disable-next-line require-atomic-updates
+    process.exitCode = 1;
```

Comment:
I think this should now live in `createSocket` too. It's doing the same checks, just different depending on whether we want to retry. This would remove some redundancy.

Reply:
Sounds reasonable, done :)

Reply:
You're still doing the same branching with repeated logic for removing the config. Also the error thrown by the recursive case in then caught and logged by the original call. 😉

Reply:
Ahahaha, my bad, now I see obvious mistake.
Moved setting status code and error logging into catch block, and left all throws in createSocket fn